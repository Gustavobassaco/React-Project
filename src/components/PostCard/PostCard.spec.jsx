import { render, screen } from '@testing-library/react';
import { PostCard } from '.';

const postMock = {
    title: 'Post Title',
    cover: 'https://via.placeholder.com/150',
};

describe('<PostCard />', () => {
    it('should render PostCard with image, title, and paragraph', () => {
        render(<PostCard post={postMock} />);

        // Verificar se a imagem foi renderizada com o src correto e o alt com o título do post
        const image = screen.getByRole('img', { name: postMock.title });
        expect(image).toBeInTheDocument();
        expect(image).toHaveAttribute('src', postMock.cover);
        expect(image).toHaveAttribute('alt', postMock.title);

        // Verificar se o título foi renderizado
        const heading = screen.getByRole('heading', { name: postMock.title });
        expect(heading).toBeInTheDocument();

        // Verificar se o parágrafo foi renderizado com o título do post
        const paragraph = screen.getByText(postMock.title, { selector: 'p' });
        expect(paragraph).toBeInTheDocument();
    });

    it('should match snapshot', () => {
        const { container } = render(<PostCard post={postMock} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
