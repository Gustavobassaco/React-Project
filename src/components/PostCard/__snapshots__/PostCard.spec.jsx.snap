// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Post Title"
    src="https://via.placeholder.com/150"
  />
  <div
    class="post-content"
  >
    <h2>
      Post Title
    </h2>
    <p>
      Post Title
    </p>
  </div>
</div>
`;
